harvester/harvester-installer
# bump rancher-monitoring chart to 102.0.0+up40.1.2

bump monitoring to 102.0.0+up40.1.2
    
      patch nginx-config file with values from 102.0.1+up40.1.2
    
      fix the tar time checking warning

      fetch rancher/shell v0.1.18, both rancher-monitoring-crd and rancher-monitoring are using it

    Signed-off-by: Jian Wang <w13915984028@gmail.com>


**Problem:**


The embedded `rancher-monitoring` in Harvester is out of dated.

**Solution:**


bump rancher-monitoring chart to` 102.0.0+up40.1.2

**Related Issue:**
https://github.com/harvester/harvester/issues/3780

**Test plan:**


<del>
note: there is an UI issue to solve, DO NOT MERGE before it is solved.  
( https://github.com/harvester/harvester/issues/3780#issuecomment-1550889891 )

cc @DaiYuzeng  @WuJun2016  @guangbochen 

update at 20230522:

In https://charts.rancher.io/index.yaml, there is `assets/rancher-monitoring/rancher-monitoring-102.0.0+up40.1.2.tgz`, but no `102.0.1+up40.1.2` , which is not formally released yet

And, in `102.0.0+up40.1.2`, there is such a bug, that is fixed in `102.0.1+up40.1.2`

https://github.com/rancher/charts/blob/dev-v2.7/charts/rancher-monitoring/102.0.1%2Bup40.1.2/charts/grafana/templates/nginx-config.yaml
```
          {{- if eq .Values.global.cattle.clusterId "local" -}}
          sub_filter '"appSubUrl":""' '"appSubUrl":"/api/v1/namespaces/{{ template "grafana.namespace" . }}/services/http:{{ template "grafana.fullname" . }}:{{ .Values.service.port }}/proxy"';
          {{- else -}}
          sub_filter '"appSubUrl":""' '"appSubUrl":"/k8s/clusters/{{ .Values.global.cattle.clusterId }}/api/v1/namespaces/{{ template "grafana.namespace" . }}/services/http:{{ template "grafana.fullname" . }}:{{ .Values.service.port }}/proxy"';
          {{- end -}}
```          
</del>

upgrade test: refer https://github.com/harvester/harvester/pull/3380

update at 20230522:

the `nginx-config` is patched, and now `rancher-monitoring` works with embedded mode in Harvester.

Test plan:
1. install new harvester with this PR
2. enable `rancher-monitoring` addon from Harvester UI
3. wait a few minutes, check cluster metrics, vm metrics on dashboard are avialable, and can click `grafana` to open embedded grafana page
